agent/structs/structs.go

```diff
@@ -384,10 +385,11 @@ func (r *ServiceSpecificRequest) CacheInfo() cache.RequestInfo {
 	// cached results, we need to be careful we maintain the same order of fields
 	// here. We could alternatively use `hash:set` struct tag on an anonymous
 	// struct to make it more robust if it becomes significant.
+	sort.Strings(r.ServiceTags)
```

Comment:
Instead of sorting this, we could do the thing the comment mentions above. That would look like.

```go
hashstructure.Hash(struct{
  name, address string
  tags []string `hash:set`
  tagFilter, connect bool
  nodeMetaFilters map[string]string
}{
  nodeMetaFilters: r.NodeMetaFilters,
  name:            r.ServiceName,	
  tags:            r.ServiceTags,			
  address:         r.ServiceAddress,
  tagFilter:       r.TagFilter,
  connect:         r.Connect,
})
```

The benefit is that this is more robust to changes in logic later, and we don't have to pay the (small) price of sorting the tags here since `hash:set` will hash them in an order-independent way.

I'm on the fence about if the extra few lines are worth it though.

For now it's still not persisted so this change has no upgrade compatibility risk and the performance difference is a micro optimisation so no big deal so let's merge it like this. This comment is kept for future reference!